sklearn/tests/test_pipeline.py

```diff
@@ -459,6 +467,26 @@ def test_predict_with_predict_params():
     assert pipe.named_steps['clf'].got_attribute
 
 
+def test_predict_proba_with_predict_params():
```

Comment:
We can update `test_predict_with_predict_params` above into the following:

```python
@pytest.mark.parametrize("method_name", [
    "predict", "predict_proba", "predict_log_proba"
])
def test_predict_methods_with_predict_params(method_name):
    # tests that Pipeline passes predict_* to the final estimator
    # when predict_* is invoked
    pipe = Pipeline([('transf', Transf()), ('clf', DummyEstimatorParams())])
    pipe.fit(None, None)
    method = getattr(pipe, method_name)
    method(X=None, got_attribute=True)

    assert pipe.named_steps['clf'].got_attribute
```

This takes advantage of `pytest.mark.parametrize` to test all the methods at once.